### PR TITLE
fix(falkordb): skip the create index error when the index is already …

### PR DESCRIPTION
**Pull request by noahxzhu**
…existing

# Description

I initialized a graph store with the falkordb. When I restart my application with the same store, I'll get an error due to the falkor index is already existing. I haven't found a better way to handle it, just use try/except to skip it, that works fine for me. 

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py [hunk 1]
```diff
@@ -31,7 +31,11 @@ def __init__(
         self._node_label = node_label
 
         self._driver = FalkorDB.from_url(url).select_graph(database)
-        self._driver.query(f"CREATE INDEX FOR (n:`{self._node_label}`) ON (n.id)")
+
+        try:
+            self._driver.query(f"CREATE INDEX FOR (n:`{self._node_label}`) ON (n.id)")
+        except Exception as e:
```
Comment (nerdai) [inline llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py:37]: what's the specific exception that you get when the index still exists? Preferably there is another way to check if the index exists versus catching an error, however, if not then I think what would be better here is to catch/pass only the specific error raised when an index already exists, and then raise any other exceptions.

Comment (noahxzhu) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py:37]: This is the error "redis.exceptions.ResponseError: Attribute 'id' is already indexed", let me think, maybe I can find an appropriate way to fix this.

Comment (noahxzhu) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py]: Hi @nerdai I didn't find a good way to solve it, it looks like it's impossible to check if the index is already existing when creating the index, if I split the check and execute with several steps, there could be a concurrency(race) issue, I just put the specific error there, and added a todo comment.

Also I reverted the legacy updated.

File: llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py [hunk 2]
```diff
@@ -31,7 +32,12 @@ def __init__(
         self._node_label = node_label
 
         self._driver = FalkorDB.from_url(url).select_graph(database)
-        self._driver.query(f"CREATE INDEX FOR (n:`{self._node_label}`) ON (n.id)")
```
Comment (logan-markewich) [inline llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py:34]: Couldn't we instead query if the index exists already? Seems a little safer (no worries if not though)

Comment (nerdai) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py:34]: @noahxzhu any thoughts on logan's suggestion above?

Comment (noahxzhu) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py:34]: @nerdai @logan-markewich did an update and tested, this should be fine for most cases, but not elegant, I'm not so familiar with falkordb.

File: llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py [hunk 3]
```diff
@@ -31,7 +32,24 @@ def __init__(
...
+        try:
+            result_set = self._driver.query(
+                f"""
+                CYPHER lbl='{self._node_label}' prop='id'
+                CALL db.indexes() YIELD label, properties
+                WHERE label = $lbl AND $prop in properties
+                RETURN 1 AS index_exists
+                """
+            ).result_set
+            # TODO: this check may cause a race issue due to these two queries are not atomic.
```
Comment (nerdai) [inline llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py:45]: Hmm I think maybe what we had before is better than potentially introducing another error/fault.

Comment (noahxzhu) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py:45]: @nerdai agree, I prefer the previous solution, @logan-markewich  what do you think? revert or keep this?

Comment (nerdai) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py:45]: Reading from his last message I'd infer that he would want to revert to previous solution as well.

Comment (noahxzhu) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-falkordb/llama_index/graph_stores/falkordb/base.py]: @nerdai reverted!

**Review by nerdai — APPROVED** [review]
thanks for digging into this @noahxzhu!

**Pull request merged**